### ISSUE TITLE
Index Not Created When Specified at Field Level When Index Type Specified (e.g. 2dsphere)

### ISSUE THREAD

**Issue by taicho** [Automattic/mongoose#6113]
Mongoose 5.0.4

**Do you want to request a *feature* or report a *bug*?**
Bug
**What is the current behavior?**
When a single sub-document schema is specified whether using an inline sub-schema declaration or referencing a schema to be used as a sub-schema if an index type is specified on the field containing the subschema/subdocument as would be used for geospatial indexes Mongoose does not create the index.

Example Schema:
```
// index on "myLocation" not created.
const schema = new Schema({
	foo: String,
	bar: Boolean,
	myLocation: {
		type: new Schema({
			coodinates: {
				type: [Schema.Types.Mixed]
			},
			type: {
				type: String
			}
		}),
		index: '2dsphere'
	},
});
const model = mongoose.model('MySchema', schema);

// Alternative Syntax
const subSchema = new Schema({
	coodinates: {
		type: [Schema.Types.Mixed]
	},
	type: {
		type: String
	}
});

// index on "myLocation" not created.
const schema = new Schema({
	foo: String,
	bar: Boolean,
	myLocation: {
		type: subSchema,
		index: '2dsphere'
	},
});
const model = mongoose.model('MySchema', schema);
```

**If the current behavior is a bug, please provide the steps to reproduce.**
1. See above for examples to replicate behavior.

**What is the expected behavior?**
A '2dsphere' schema should be created on the "MySchema" collection for the "myLocation" field.

MongoDB: 3.4.4
Mongoose: 5.0.4
Node: 8.1.2

**Comment by vkarpov15** [comment]
Thanks for reporting, will investigate asap

**Comment by vkarpov15** [comment]
Fixed, fix will be in 5.0.7 release :+1:
